### PR TITLE
coqPackages.coq-lsp: 0.1.7 -> 0.1.8

### PR DESCRIPTION
**Pull request by Zimmi48**
## Description of changes

cc @Alizter @ejgallego FYI

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/coq-modules/coq-lsp/default.nix [hunk 1]
```diff
@@ -10,12 +10,15 @@ mkCoqDerivation rec {
   release."0.1.7+8.16".sha256 = "sha256-ZBxwrnnCmT5q4C7ocQ+M+aSJQNnEjeN2HFw4bzPozYs=";
   release."0.1.7+8.17".sha256 = "sha256-f671wzGQannGjRbmBRHFKXz24BTPX7oVeHUxnv4Vd6Y=";
   release."0.1.7+8.18".sha256 = "sha256-J+bRIzjdIPRu7DvAGVBKB43O3UJliTo8XQ87OTzsFyc=";
+  release."0.1.8+8.16".sha256 = "sha256-dEEAK5IXGjHB8D/fYJRQG/oCotoXJuWLxXB0GQlY2eo=";
+  release."0.1.8+8.17".sha256 = "sha256-TmaE+osn/yAPU1Dyni/UTd5w/L2+qyPE3H/g6IWvHLQ=";
+  release."0.1.8+8.18".sha256 = "sha256-UYmiDdbax4wxp5dLia/1t1gFyK6UELtJJvDMd5Hd14s=";
```
Comment (Alizter) [inline pkgs/development/coq-modules/coq-lsp/default.nix:15]: What's the point of keeping the older versions around if we don't use them?

Comment (Alizter) [reply pkgs/development/coq-modules/coq-lsp/default.nix:15]: Sorry I should have highlighted the 3 lines above these.

Comment (Alizter) [inline pkgs/development/coq-modules/coq-lsp/default.nix:15]:
```suggestion
  release."0.1.8+8.16".sha256 = "sha256-dEEAK5IXGjHB8D/fYJRQG/oCotoXJuWLxXB0GQlY2eo=";
  release."0.1.8+8.17".sha256 = "sha256-TmaE+osn/yAPU1Dyni/UTd5w/L2+qyPE3H/g6IWvHLQ=";
  release."0.1.8+8.18".sha256 = "sha256-UYmiDdbax4wxp5dLia/1t1gFyK6UELtJJvDMd5Hd14s=";
```

**Comment by ejgallego** [comment]
Thanks a lot @Zimmi48 @Alizter !

Question: several Coq versions have bugs that while fixed in our `lsp` branches, didn't arrive to Coq upstream. In particular:
- problems working with multiple files: fixed in Coq 8.17.1, lower versions need branch with fix
- problems due to async interruptions: fixed in `master`, hopefully in 8.18.1 too

There bugs are serious and actually affect all document managers; is it possible to tell Nix to build `coq-lsp` using one of the branches that have been fixed?

File: pkgs/development/coq-modules/coq-lsp/default.nix (hunk 1 above)
Comment (Zimmi48) [reply pkgs/development/coq-modules/coq-lsp/default.nix:15]: It's kind of the habit in the `coqPackages` realm, but I agree that for a tool like this, there is no real point in keeping them.

**Comment by Zimmi48** [comment]
Do these bug fixes correspond to simple patches? Because one easy way to do that would be to add them as patches to the Coq derivation directly (if other maintainers, such as @vbgl agrees).

Alternatively, if only coq-lsp needs these patches, but coq-lsp built with a patched Coq would still be able to load libraries built with an unpatched Coq, we could specifically override the Coq dependency in the coq-lsp derivation (which means that people using coq-lsp would build / download two versions of Coq instead of just one...).

**Comment by ejgallego** [comment]
> Do these bug fixes correspond to simple patches?

Yes, for 8.17 / 8.18 is a single patch, for 8.16 we need two patches.

> Because one easy way to do that would be to add them as patches to the Coq derivation directly (if other maintainers, such as @vbgl agrees).

I don't see any downside to that, the patch is pretty straightforward, correcting a few places where exceptions should have not been absorbed. This patch is needed for all interfaces + tactician, without it most interfaces crash the moment to press "Interrupt" while you are using some of the affected tactics. (coq-lsp doesn't crash, but wrongly believes the tactic failed, whereas it was just interrupted)

Happy to go this way, that's the patch https://github.com/coq/coq/pull/17638

Another option is to release 8.17.2 and 8.18.1 .

> Alternatively, if only coq-lsp needs these patches, but coq-lsp built with a patched Coq would still be able to load libraries built with an unpatched Coq, we could specifically override the Coq dependency in the coq-lsp derivation (which means that people using coq-lsp would build / download two versions of Coq instead of just one...).

The patches shouldn't affect that capability to load libraries, but indeed it would be safer to rebuild the libs with the new coq. How does that work tho? In the sense that imagine we have two derivations that take coq as an input, let's say `math-comp` and `coq-lsp`.

It is allowed for the inputs to differ?

**Comment by Alizter** [comment]
I think the easiest way would be to add the patches to the Coq derivation directly so that everybody using Coq from nixpkgs can benefit from it. If we keep two version of Coq, then things will spiral into complication very quickly.

**Comment by Zimmi48** [comment]
OK, let's add the patches to the Coq derivation, then, but in another PR.

**Pull request merged**